custom/mvp_ucr/reports/expressions.py

```diff
@@ -0,0 +1,205 @@
+
+        if parent_node:
+            if 'q1006_1' in parent_node and int(parent_node['q1006_1']) == 1:
+                treatment_providers += "CHW at home <br>"
+            if 'q1006_2' in parent_node and int(parent_node['q1006_2']) == 1:
+                treatment_providers += "Friend/Relative at home <br>"
+            if 'q1006_3' in parent_node and int(parent_node['q1006_3']) == 1:
+                treatment_providers += "Traditional healer <br>"
+            if 'q1006_4' in parent_node and int(parent_node['q1006_4']) == 1:
+                treatment_providers += "Health clinic/Post <br>"
+            if 'q1006_5' in parent_node and int(parent_node['q1006_5']) == 1:
+                treatment_providers += "Hospital <br>"
+            if 'q1006_6' in parent_node and int(parent_node['q1006_6']) == 1:
+                treatment_providers += "Pharmacy/drug seller/store <br>"
+            if 'q1006_7' in parent_node and int(parent_node['q1006_7']) == 1:
+                if 'q1006_96' in parent_node:
+                    treatment_providers += parent_node['q1006_96']
+                if 'q1006_7_96' in parent_node:
+                    treatment_providers += parent_node['q1006_7_96']
+
+        if form == ADULT_FORM:
+            adult_answer = ""
+            try:
+                treatment_providers_choices = doc['interview']['q1006_q1060']['q1006']
+            except KeyError:
+                pass
+
+            if len(adult_answer) > 0:
+                choices = treatment_providers_choices.split(" ")
+                for x in choices:
+                    if int(x) == 1:
+                        treatment_providers += "CHW at home <br>"
+                    if int(x) == 2:
+                        treatment_providers += "Friend/Relative at home <br>"
+                    if int(x) == 3:
+                        treatment_providers += "Traditional healer <br>"
+                    if int(x) == 4:
+                        treatment_providers += "Health clinic/Post <br>"
+                    if int(x) == 5:
+                        treatment_providers += "Hospital <br>"
+                    if int(x) == 6:
+                        treatment_providers += "Pharmacy/drug seller/store <br>"
+                    if int(x) == 7:
+                        try:
+                            treatment_providers += doc['interview']['q1006_q1060']['q1006_7']
+                        except KeyError:
+                            pass
+        return treatment_providers
+
+    def __call__(self, item, context=None):
+
+        docs = []
+
+        docs.append({'mvp_death_place': self._death_place(item['xmlns'], item['form']),
+                    'mvp_death_cause': self._death_cause(item['xmlns'], item['form']),
+                    'mvp_treatment_place_name': self._treatment_place_name(item['xmlns'], item['form']),
+                    'mvp_treatment_providers': self._treatment_providers(item['xmlns'], item['form'])})
```

Comment:
@NoahCarnahan can you clarify if you meant something like this?

```
class TreatmentProviderGetterSpec(JsonObject):
    type = TypeProperty('treatment_provider_name')
    treatment_provider_name = DefaultProperty(required=False)

    def __call__(self, item, context=None):
        #formatTreatmentProviderList
        return FormatedListOFTreatmentProvider

class MedicalCauseGetterSpec(JsonObject):
    type = TypeProperty('medical_cause')
    medical_cause = DefaultProperty(required=False)

    def __call__(self, item, context=None):
        #formatTreatmentProviderList
        return FormatedListOFTreatmentProvider

```

and then:

{
            "type": "expression",
            "expression": {
                    "type": "treatment_provider_name",
                    "treatment_provider_name":  ""
            },
            "column_id": "treatment_provider",
            "datatype": "string",
            "display_name": "Preatment Provider"
        }